testing/trees.R

```diff
@@ -0,0 +1,289 @@
+# Alternative would be to use c4.5 trees instead of CART.
+
+wrapperRF <- function(type, y, train, y.test = NULL, test = NULL, tune = F, ordered.vars = NULL, 
+                       probability = F,
+                       num.trees = 500, 
+                       mtry = NULL,
+                       importance = "impurity_corrected",
+                       quantreg = F,
+                       num.threads = as.numeric(Sys.getenv("NUMBER_OF_PROCESSORS")), 
+                       ...) {
+  
+  train[ordered.vars] <- lapply(train[ordered.vars],
+                                function(x) if(!is.ordered(x)) factor(x, ordered = T) else x)
+  
+  if(type == "surv") {
+    
+    statvar = y[[1]] 
+    if("time" %in% y) depvar = y[["time"]] else depvar = NULL
+    
+  } else {
+    
+    depvar = y[[1]]
+    statvar = NULL
+    
+  }
+  
+  if(is.null(mtry)) {
+    
+    ncol <- ncol(train) # Proxy for number of predictors (not perfect)
+    
+    mtry = floor(switch(type,
+                        "classif" = sqrt(ncol - 1),
+                        "regr" = (ncol - 1)/3,
+                        "surv" = if("time" %in% y) sqrt(ncol - 2) else sqrt(ncol - 1)))
+    
+  } else mtry = floor(mtry)
+  
+  if(!tune) {
+    
+    fit <- ranger(data = train,
+                  dependent.variable.name = depvar,
+                  status.variable.name = statvar,
+                  probability = probability,
+                  importance = importance, scale.permutation.importance = F,
+                  num.trees = num.trees,
+                  mtry = mtry,
+                  quantreg = F,
+                  num.threads = num.threads,
+                  respect.unordered.factors = T,
+                  write.forest = T,
+                  ...)
+    
+  } else {
+    
+    task.funs = c("makeClassifTask", "makeRegrTask") # tuneRanger does not appear
+    
+    task = do.call(tasks[grepl("regr", task.funs, ignore.case = T)], list(data = train, target = y[[1]]))
+    
+    estimateTimeTuneRanger(iris.task) # Estimated runtime to be displayed to the user
+    
+    allArgs <- c(mget(names(formals()), sys.frame(sys.nframe())), list(...))
+    
+    # Tune mtry and sampling scheme (sample size and replacement)
+    res = tuneRanger(task, 
+                     parameters = allArgs[formalArgs(ranger)],
+                     tune.parameters = c("mtry", "replace", "sample.fraction"), 
+                     num.trees = num.trees,
+                     num.threads = num.threads, 
+                     iters = 70, iters.warmup = 30)
+  }
+  
+  #Performance summary  
+  if(!is.null(test)) {
+    
+    testError(model = fit, y.test = y.test, test = test)
+    
+  }
+  
+  # Variable importance plots
+  
+  # p-values (need to look into holdoutRF as well)
+  pval = importance_pvalues(fit, method = "altmann")
+  
+  # ICE plots
+  
+  
```

Comment:
```
z <- list(response = list(...), coefs = ..., ...)
class(z) <- c("inz.tree.ranger", "inz.tree", "inz.mbuilder")
z
```

Reply:
Alternatively ... (probably better/easier)
```suggestion
z <- list(x = the_ranger_thing, ...)
class(z) <- "inz.mbuilder"
z
```

Then you'll create a method ... 
```
summary.inz.mbuilder <- function(object, ...) {
  summary(z$x)
}
```